src/Libraries/DSCPython/CPythonEvaluator.cs

```diff
@@ -236,6 +236,20 @@ static CPythonEvaluator()
             }
         }
 
+        /// <summary>
+        /// Creates and initializaes the global Python scope.
+        /// </summary>
+        private static PyScope CreateGlobalScope()
+        {
+            var scope = Py.CreateScope(globalScopeName);
+            // Allows discoverability of modules by inspecting their attributes
+            scope.Exec(@"
```

Comment:
I'm confused why this only needed to be run on this scope...
I guess it may be related to the `sys.path` issue @reddyashish is looking at and some confusion I have around the lifetime of ironPython engines vs PythonNet ones.

Reply:
Yes. From what I could see, loaded modules stay alive beyond a single Python node evaluation. It's like they stay alive with the process. Doing it here makes more sense `clr.setPreload(True)`, not only because it's easier, but also because once a module is loaded you can't reload it.